src/Agent.Worker/Build/TfsVCSourceProvider.cs

```diff
@@ -596,6 +582,21 @@ public string NormalizedServerPath
             public bool Recursive => !(ServerPath ?? string.Empty).EndsWith("/*");
 
             public string ServerPath { get; set; }
+
+            /// <summary>
+            /// Gets the local path specified in the definition to a place on disk and normalizes slashes.
+            /// </summary>
+            public string GetRootedLocalPath(string sourcesDirectory)
```

Comment:
In addition to moving this method, I changed the logic up. I need to do E2E testing. The old logic looks wrong to me (converted `\` to `/` on nix). I don't know why I previously felt like that was appropriate. I need to do E2E testing and confirm that it's not a limitation of TFS. If it is, then I'll revert and leave a big comment in the source code.